### PR TITLE
PF4: Drawer updates for Master/Detail Support

### LINKED ISSUE #3609: Drawer: updates to support master/detail layout
https://github.com/patternfly/patternfly-next/pull/2520/files

* adds an element `.pf-c-drawer__content-body` inside of `.pf-c-drawer__content` to hold the content passed to the `__content` element
* adds `aria-hidden="true"` to the `__panel` element if the drawer is not expanded
* `<DrawerContent>` should support a prop to add padding to the content, which applies `.pf-m-padding` to `.pf-c-drawer__content-body`
* `<Drawer>` should support...
  * prop to position the panel to slide out from the left by applying `.pf-m-panel-left` to `.pf-c-drawer`
  * prop to use a line as the divider between the panel and content (instead of a box shadow) by applying `.pf-m-divider` to `.pf-c-drawer`

cc @mattnolting

### PR DESCRIPTION
**Pull request by tlabaj**
**What**: Closes #3609
Core version was updated to "@patternfly/patternfly": "2.66.0" in the PR to pull in drawer changes.  The core update would break the build due to some classes that were removed.  The change in the PR are needed for the upgrade. 

**Additional issues**:
TODO:
- [ ] - Add functionality for setting panel width
- [ ] - Update Demo App
- [ ] - Update Integration Test
(Issue #3888 has been opened to track the follow up work)

### CODE REVIEW

File: packages/react-table/package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
   },
   "homepage": "https://github.com/patternfly/patternfly-react/tree/master/packages/react-table#readme",
   "dependencies": {
-    "@patternfly/patternfly": "2.65.3",
+    "@patternfly/patternfly": "2.66.0",,
```
Comment (evwilkin) [inline packages/react-table/package.json]: Extra comma

File: packages/react-core/src/components/Drawer/DrawerActions.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Drawer/drawer';
+import { css } from '@patternfly/react-styles';
+
+export interface DrawerActionsProps extends React.HTMLProps<HTMLDivElement> {
+  /** Additional classes added to the drawer actions button. */
+  className?: string;
+  /** Actions to rendered in the panel head. */
```
Comment (evwilkin) [inline packages/react-core/src/components/Drawer/DrawerActions.tsx]: Nit - "Actions to **be** rendered..."

File: packages/react-core/src/components/Drawer/DrawerActions.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  children?: React.ReactNode;
+}
+
+export const DrawerActions: React.SFC<DrawerActionsProps> = ({
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  className = '',
+  children,
+  ...props
+}: DrawerActionsProps) => (
+  <div className={css(styles.drawerActions)} {...props}>
```
Comment (evwilkin) [inline packages/react-core/src/components/Drawer/DrawerActions.tsx]: Do we need to pass in className from props here?

Comment (tlabaj) [reply packages/react-core/src/components/Drawer/DrawerActions.tsx]: yes!  Good catch!

File: packages/react-core/src/components/Drawer/DrawerCloseButton.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+export const DrawerCloseButton: React.SFC<DrawerCloseButtonProps> = ({
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  className = '',
+  onClose = () => undefined as any,
+  'aria-label': ariaLabel = 'Close drawer panel',
+  ...props
+}: DrawerCloseButtonProps) => (
+  <div className={css(styles.drawerClose)} {...props}>
```
Comment (evwilkin) [inline packages/react-core/src/components/Drawer/DrawerCloseButton.tsx]: I believe we need to pass in className from props here as well

File: packages/react-core/src/components/Drawer/DrawerContent.tsx [hunk 5]
```diff
@@ -1,21 +1,28 @@
...
 import styles from '@patternfly/react-styles/css/components/Drawer/drawer';
 import { css } from '@patternfly/react-styles';
+import { DrawerMain } from './DrawerMain';
 
 export interface DrawerContentProps extends React.HTMLProps<HTMLDivElement> {
   /** Additional classes added to the Drawer. */
   className?: string;
   /** Content to rendered in the drawer */
   children?: React.ReactNode;
+  /** Content rendered in drawer the panel */
```
Comment (evwilkin) [inline packages/react-core/src/components/Drawer/DrawerContent.tsx]: Nit - "Content rendered *in the drawer* panel"

File: packages/react-core/src/components/Drawer/DrawerContent.tsx [hunk 6]
```diff
@@ -1,21 +1,28 @@
...
   className = '',
   children,
+  panelContent,
   ...props
 }: DrawerContentProps) => (
-  <div className={css(styles.drawerContent)} {...props}>
-    {children}
-  </div>
+  <DrawerMain>
+    <div className={css(styles.drawerContent)} {...props}>
```
Comment (evwilkin) [inline packages/react-core/src/components/Drawer/DrawerContent.tsx]: I believe we need to pass in className from props here

File: packages/react-core/src/components/Drawer/DrawerContentBody.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+export const DrawerContentBody: React.SFC<DrawerContentBodyProps> = ({
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  className = '',
+  children,
+  hasPadding = false,
+  ...props
+}: DrawerContentBodyProps) => (
+  <div className={css(styles.drawerBody, hasPadding && styles.modifiers.padding)} {...props}>
```
Comment (evwilkin) [inline packages/react-core/src/components/Drawer/DrawerContentBody.tsx]: I believe we need to pass through className from props here

File: packages/react-core/src/components/Drawer/DrawerHead.tsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+export const DrawerHead: React.SFC<DrawerHeadProps> = ({
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  className = '',
+  children,
+  noPadding = false,
+  ...props
+}: DrawerHeadProps) => (
+  <DrawerPanelBody noPadding={noPadding}>
+    <div className={css(styles.drawerHead)} {...props}>
```
Comment (evwilkin) [inline packages/react-core/src/components/Drawer/DrawerHead.tsx]: I believe we have to pass through className from props here

File: packages/react-core/src/components/Drawer/DrawerMain.tsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  children?: React.ReactNode;
+}
+
+export const DrawerMain: React.SFC<DrawerMainProps> = ({
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  className = '',
+  children,
+  ...props
+}: DrawerMainProps) => (
+  <div className={css(styles.drawerMain)} {...props}>
```
Comment (evwilkin) [inline packages/react-core/src/components/Drawer/DrawerMain.tsx]: I believe we have to pass through className from props here

File: packages/react-core/src/components/Drawer/DrawerPanelBody.tsx [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+export const DrawerPanelBody: React.SFC<DrawerPanelBodyProps> = ({
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  className = '',
+  children,
+  noPadding = false,
+  ...props
+}: DrawerPanelBodyProps) => (
+  <div className={css(styles.drawerBody, noPadding && styles.modifiers.noPadding)} {...props}>
```
Comment (evwilkin) [inline packages/react-core/src/components/Drawer/DrawerPanelBody.tsx]: I believe we have to pass through className from props here

File: packages/react-core/src/components/Drawer/DrawerSection.tsx [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+  children?: React.ReactNode;
+}
+
+export const DrawerSection: React.SFC<DrawerSectionProps> = ({
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  className = '',
+  children,
+  ...props
+}: DrawerSectionProps) => (
+  <div className={css(styles.drawerSection)} {...props}>
```
Comment (evwilkin) [inline packages/react-core/src/components/Drawer/DrawerSection.tsx]: I believe we have to pass through className from props here

**Pull request merged**